lib/dry/configurable.rb

```diff
@@ -114,14 +115,30 @@ def _settings
     def _config_for(&block)
       config_klass = ::Class.new { extend ::Dry::Configurable }
       config_klass.instance_eval(&block)
-      config_klass.config
+      create_nested_config(config_klass)
```

Comment:
Awesome thats great I will update the code, and the missing test so we can move forward